src/rlookup.c

```diff
@@ -999,7 +1029,11 @@ static int RLookup_JSON_GetAll(RLookup *it, RLookupRow *dst, RLookupLoadOptions
   if (res == REDISMODULE_ERR) {
     goto done;
   }
-  RLookupKey *rlk = RLookup_GetKeyEx_TEMP(it, JSON_ROOT, strlen(JSON_ROOT), RLOOKUP_F_OCREAT);
+  RLookupKey *rlk = RLookup_FindKey(it, JSON_ROOT, strlen(JSON_ROOT));
+  if (!rlk) {
+    // First returned document, create the key.
+    rlk = RLookup_GetKey_LoadEx(it, JSON_ROOT, strlen(JSON_ROOT), JSON_ROOT, RLOOKUP_F_NOFLAGS);
```

Comment:
Same question on avoiding `RLookup_FindKey`

Reply:
A more segnificant optimization for JSON will be to have a "single" field load of the root, and avoid serching for it even once for every doc